packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -1089,6 +1102,25 @@ export class StripeHelper {
         continue;
       }
 
+      const result = subscriptionProductMetadataValidator.validate({
+        ...item.product.metadata,
+        ...item.metadata,
+      });
+
+      if (result?.error) {
+        const msg = `fetchAllPlans - Plan "${item.id}"'s metadata failed validation`;
+        this.log.error(msg, { item: item, error: result.error });
```

Comment:
Use 'plan' instead of 'item' here for the key, since the object is a plan?  ('item' is just the name of the variable.)